### ISSUE TITLE
[v9.0.0-rc.2] Event props in useSprings props function are not saved for future updates

### ISSUE THREAD

**Issue by aleclarson** [react-spring/react-spring#997]
In v9-beta.34, I could pass `onFrame` to useSpring like this
```jsx
const [props, set] = useSprings(pages.length, i => ({
    x: i * window.innerWidth,
    scale: 1,
    display: 'block',
    onFrame: f => console.log(f),
    onRest: f => console.log(f)
}))
```
I've updated `onFrame` -> `onChanged`, but neither `onChange` or `onRest` seem to get called
```jsx
const [props, set] = useSprings(pages.length, i => ({
    x: i * window.innerWidth,
    scale: 1,
    display: 'block',
    onChange: f => console.log(f),
    onRest: f => console.log(f)
}))
````

repro: https://codesandbox.io/s/viewpager-s2ukf?file=/src/index.js

_Originally posted by @tim-soft in https://github.com/react-spring/react-spring/issues/985#issuecomment-626221266_

**Comment by tim-soft** [comment]
I don't have a demo for this, but I've noticed the same behavior in the `useSpring` hook

**Comment by aleclarson** [comment]
Fixed in `9.0.0-rc.3`
